shared/js/background/events.es6.js

```diff
@@ -183,75 +183,79 @@ browser.webRequest.onBeforeRequest.addListener(
-    }
+const extraInfoSpec = manifestVersion === 3 ? ['responseHeaders'] : ['blocking', 'responseHeaders']
 
-    // We determine if browsingTopics is enabled by testing for availability of its
-    // JS API.
-    // Note: This approach will not work with MV3 since the background
-    //       ServiceWorker does not have access to a `document` Object.
-    const isTopicsEnabled = ('browsingTopics' in document) && utils.isFeatureEnabled('googleRejected')
-    browser.webRequest.onHeadersReceived.addListener(
-        request => {
-            if (request.type === 'main_frame') {
-                tabManager.updateTabUrl(request)
-
-                const tab = tabManager.get({ tabId: request.tabId })
-                // SERP ad click detection
-                if (
-                    utils.isRedirect(request.statusCode)
-                ) {
-                    tab.setAdClickIfValidRedirect(request.url)
-                } else if (tab && tab.adClick && tab.adClick.adClickRedirect && !utils.isRedirect(request.statusCode)) {
-                    tab.adClick.adClickRedirect = false
-                    tab.adClick.adBaseDomain = tab.site.baseDomain
-                }
-            }
+if (browser.webRequest.OnHeadersReceivedOptions.EXTRA_HEADERS) {
+    extraInfoSpec.push(browser.webRequest.OnHeadersReceivedOptions.EXTRA_HEADERS)
+}
 
-            if (ATB.shouldUpdateSetAtb(request)) {
-                // returns a promise
-                return ATB.updateSetAtb()
-            }
+// We determine if browsingTopics is enabled by testing for availability of its
+// JS API.
+// Note: This approach will not work with MV3 since the background
+//       ServiceWorker does not have access to a `document` Object.
+const isTopicsEnabled = manifestVersion === 3 ? false : ('browsingTopics' in document) && utils.isFeatureEnabled('googleRejected')
 
-            const responseHeaders = request.responseHeaders
+browser.webRequest.onHeadersReceived.addListener(
+    request => {
+        if (request.type === 'main_frame') {
+            tabManager.updateTabUrl(request)
 
-            if (isTopicsEnabled && responseHeaders && (request.type === 'main_frame' || request.type === 'sub_frame')) {
-                // there can be multiple permissions-policy headers, so we are good always appending one
-                // According to Google's docs a site can opt out of browsing topics the same way as opting out of FLoC
-                // https://privacysandbox.com/proposals/topics (See FAQ)
-                responseHeaders.push({ name: 'permissions-policy', value: 'interest-cohort=()' })
+            const tab = tabManager.get({ tabId: request.tabId })
+            // SERP ad click detection
+            if (
+                utils.isRedirect(request.statusCode)
+            ) {
+                tab.setAdClickIfValidRedirect(request.url)
+            } else if (tab && tab.adClick && tab.adClick.adClickRedirect && !utils.isRedirect(request.statusCode)) {
```

Comment:
Nit: Could use optional properties?
```suggestion
            } else if (tab && tab.?adClick.?adClickRedirect && !utils.isRedirect(request.statusCode)) {
```